helloworld/personality/personality.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Runs a simple Trillian personality.
+package personality
+
+import (
+	"context"
+	"flag"
+	"log"
+	"time"
+
+	"github.com/google/trillian"
+	"github.com/google/trillian/merkle/rfc6962/hasher"
+	tt "github.com/google/trillian/types"
+	"google.golang.org/grpc"
+)
+
+var (
+	connectTimeout = flag.Duration("connect_timeout", 5*time.Second, "the timeout for connecting to the backend")
+)
+
+type Chkpt struct {
+	LogSize  int64
+	RootHash []byte
+}
+
+type TrillianP struct {
+	l      trillian.TrillianLogClient
+	treeID int64
+}
+
+// NewPersonality creates a new Trillian personality from the flags.
+func NewPersonality(logAddr string, treeID int64) TrillianP {
+	if treeID <= 0 {
+		log.Fatalf("tree_id must be provided and positive, got %d", treeID)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), *connectTimeout)
+	defer cancel()
+	conn, err := grpc.DialContext(ctx, logAddr, grpc.WithInsecure(), grpc.WithBlock())
+	if err != nil {
+		panic(err)
+	}
+
+	log := trillian.NewTrillianLogClient(conn)
+
+	return TrillianP{
+		l:      log,
+		treeID: treeID,
+	}
+}
+
+// formLeaf creates a Trillian log leaf from an entry.
+func (p TrillianP) formLeaf(entry []byte) *trillian.LogLeaf {
+	leafHash := hasher.DefaultHasher.HashLeaf(entry)
+	return &trillian.LogLeaf{
+		LeafValue:      entry,
+		MerkleLeafHash: leafHash,
+	}
+}
+
+// GetChkpt gets the latest checkpoint.
+func (p TrillianP) GetChkpt(ctx context.Context) Chkpt {
+	req := trillian.GetLatestSignedLogRootRequest{LogId: p.treeID}
+	resp, err := p.l.GetLatestSignedLogRoot(ctx, &req)
+	if err != nil {
+		panic(err)
+	}
+	root := resp.GetSignedLogRoot()
+	var logRoot tt.LogRootV1
+	if err := logRoot.UnmarshalBinary(root.LogRoot); err != nil {
+		panic(err)
+	}
```

Comment:
```suggestion
        // Unpack the response and convert it to the local Checkpoint representation
	root := resp.GetSignedLogRoot()
	var logRoot tt.LogRootV1
	if err := logRoot.UnmarshalBinary(root.LogRoot); err != nil {
		panic(err)
	}
```

Prefer to comment logical blocks like this to signal the intent. This allows the example code to be more easily digested and scanned over.